app/controllers/tax_returns_controller.rb

```diff
@@ -0,0 +1,54 @@
+class TaxReturnsController < ApplicationController
+  # Because these aren't CRUD verb methods, they need to be explicitly listed on the before actions
+  before_action :block_access_on_production, only: [:authorize_signature, :sign, :success]
+  before_action :load_tax_return, except: [:success]
+  before_action :check_for_forms, only: [:sign, :authorize_signature]
+
+  def authorize_signature
+    tax_return = TaxReturn.find(params[:tax_return_id])
+    @form = Portal::SignForm8879.new(tax_return)
+  end
+
+  def sign
+    @form = Portal::SignForm8879.new(@tax_return, permitted_params)
+    if @form.valid? && @form.sign
+      redirect_to tax_return_success_path(params[:tax_return_id])
+    else
+      flash.now[:alert] = I18n.t("controllers.tax_returns_controller.errors.#{@form.errors.keys.first}")
+      render :authorize_signature
+    end
+  end
+
+  def success; end
+
+  private
+
+  def load_tax_return
+    @tax_return = TaxReturn.find(params[:tax_return_id])
+  end
+
+  def permitted_params
+    params
+      .require(Portal::SignForm8879.form_param)
```

Comment:
neat! i didn't know about this

Reply:
Yes! I added this a while back to make working with form objects a bit easier.